sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ConversationAsyncClient.java

```diff
@@ -469,17 +486,199 @@
+     * @param override - True to override the file if it exists.
+     * @return An empty {@link Mono} object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> downloadTo(
+        String path,
+        URI endpoint,
+        HttpRange range,
+        ParallelDownloadOptions parallelDownloadOptions,
+        boolean override) {
+        return downloadToWithResponse(path, endpoint, range, parallelDownloadOptions, override)
+            .map(Response::getValue);
+    }
+
+    /**
+     * Download the recording content (e.g. Recording's metadata, Recording video, etc.) from the {@code endpoint}
+     * and stored into the file in {@code path}.
+     * @param path - File path to store file to.
+     * @param endpoint - Location of the recording's content.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @param parallelDownloadOptions - An optional {@link ParallelDownloadOptions} object to modify how the
+     *                               parallel download will work.
+     * @param override - True to override the file if it exists.
+     * @return A {@link Mono} object containing the http response information from the download.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> downloadToWithResponse(
+        String path,
+        URI endpoint,
+        HttpRange range,
+        ParallelDownloadOptions parallelDownloadOptions,
+        boolean override) {
+        return downloadToWithResponse(path, endpoint, range, parallelDownloadOptions, override, null);
+    }
+
+    Mono<Response<Void>> downloadToWithResponse(
+        String path,
+        URI endpoint,
+        HttpRange range,
+        ParallelDownloadOptions parallelDownloadOptions,
+        boolean override,
+        Context context) {
+        HttpRange finalRange = range == null ? new HttpRange(0) : range;
+        ParallelDownloadOptions finalParallelDownloadOptions =
+            parallelDownloadOptions == null
+                ? new ParallelDownloadOptions()
+                : parallelDownloadOptions;
+        Set<OpenOption> openOptions = new HashSet<>();
+
+        if (override) {
+            openOptions.add(StandardOpenOption.CREATE);
+        } else {
+            openOptions.add(StandardOpenOption.CREATE_NEW);
+        }
+        openOptions.add(StandardOpenOption.WRITE);
+        openOptions.add(StandardOpenOption.READ);
+
+        try {
+            AsynchronousFileChannel file = AsynchronousFileChannel.open(Paths.get(path), openOptions, null);
+            return Mono.just(file).flatMap(
+                c -> contentDownloader.downloadToFile(c, endpoint, finalRange, finalParallelDownloadOptions, context))
+                .doFinally(signalType -> contentDownloader.downloadToFileCleanup(file, path, signalType));
```

Comment:
I'd recommend the usage of `Mono.using` here: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#using-java.util.concurrent.Callable-java.util.function.Function-java.util.function.Consumer-

Reply:
The only issue with this is the `SignalType` object provided by `doFinally`. Is there any other way to get it?